rules/DowngradePhp70/Rector/Spaceship/DowngradeSpaceshipRector.php

```diff
@@ -94,17 +92,11 @@ public function refactor(Node $node): FuncCall
         $ternary = new Ternary($smaller, $ternaryIf, $ternaryElse);
         $anonymousFunction->stmts[1] = new Return_($ternary);
 
-        $currentStatement = $node->getAttribute(AttributeKey::CURRENT_STATEMENT);
-        if (! $currentStatement instanceof Node) {
-            throw new ShouldNotHappenException();
-        }
+        $variableAssignName = $this->namedVariableFactory->createVariable($node, 'battleShipcompare');
 
-        $scope = $currentStatement->getAttribute(AttributeKey::SCOPE);
-
-        $variableAssignName = $this->variableNaming->createCountedValueName('battleShipcompare', $scope);
         $variableAssign = new Variable($variableAssignName);
```

Comment:
can directly assign  

```suggestion
        $variableAssign = $this->namedVariableFactory->createVariable($node, 'battleShipcompare');
```

and remove below on line 97

Reply:
Just noticed :+1: